sentry-android-core/src/main/java/io/sentry/android/core/SentryLogcatAdapter.java

```diff
@@ -44,73 +48,104 @@ private static void addAsBreadcrumb(
+    final @NotNull ScopesAdapter scopes = ScopesAdapter.getInstance();
+    if (tr == null) {
+      scopes.logger().log(level, msg);
+    } else {
+      StringWriter sw = new StringWriter(256);
+      PrintWriter pw = new PrintWriter(sw, false);
+      tr.printStackTrace(pw);
+      pw.flush();
+      scopes.logger().log(level, msg != null ? (msg + "\n" + sw.toString()) : sw.toString());
+      pw.close();
+    }
+  }
+
   public static int v(@Nullable String tag, @Nullable String msg) {
     addAsBreadcrumb(tag, SentryLevel.DEBUG, msg);
+    addAsLog(SentryLogLevel.TRACE, msg, null);
     return Log.v(tag, msg);
   }
 
   public static int v(@Nullable String tag, @Nullable String msg, @Nullable Throwable tr) {
     addAsBreadcrumb(tag, SentryLevel.DEBUG, msg, tr);
+    addAsLog(SentryLogLevel.TRACE, msg, tr);
     return Log.v(tag, msg, tr);
   }
 
   public static int d(@Nullable String tag, @Nullable String msg) {
     addAsBreadcrumb(tag, SentryLevel.DEBUG, msg);
+    addAsLog(SentryLogLevel.DEBUG, msg, null);
     return Log.d(tag, msg);
   }
 
   public static int d(@Nullable String tag, @Nullable String msg, @Nullable Throwable tr) {
     addAsBreadcrumb(tag, SentryLevel.DEBUG, msg, tr);
+    addAsLog(SentryLogLevel.DEBUG, msg, tr);
     return Log.d(tag, msg, tr);
   }
 
   public static int i(@Nullable String tag, @Nullable String msg) {
     addAsBreadcrumb(tag, SentryLevel.INFO, msg);
+    addAsLog(SentryLogLevel.INFO, msg, null);
     return Log.i(tag, msg);
   }
 
   public static int i(@Nullable String tag, @Nullable String msg, @Nullable Throwable tr) {
     addAsBreadcrumb(tag, SentryLevel.INFO, msg, tr);
+    addAsLog(SentryLogLevel.INFO, msg, tr);
     return Log.i(tag, msg, tr);
   }
 
   public static int w(@Nullable String tag, @Nullable String msg) {
     addAsBreadcrumb(tag, SentryLevel.WARNING, msg);
+    addAsLog(SentryLogLevel.WARN, msg, null);
     return Log.w(tag, msg);
   }
 
   public static int w(@Nullable String tag, @Nullable String msg, @Nullable Throwable tr) {
     addAsBreadcrumb(tag, SentryLevel.WARNING, msg, tr);
+    addAsLog(SentryLogLevel.WARN, msg, tr);
     return Log.w(tag, msg, tr);
   }
 
   public static int w(@Nullable String tag, @Nullable Throwable tr) {
     addAsBreadcrumb(tag, SentryLevel.WARNING, tr);
+    addAsLog(SentryLogLevel.WARN, null, tr);
     return Log.w(tag, tr);
   }
 
   public static int e(@Nullable String tag, @Nullable String msg) {
     addAsBreadcrumb(tag, SentryLevel.ERROR, msg);
+    addAsLog(SentryLogLevel.ERROR, msg, null);
     return Log.e(tag, msg);
   }
 
   public static int e(@Nullable String tag, @Nullable String msg, @Nullable Throwable tr) {
     addAsBreadcrumb(tag, SentryLevel.ERROR, msg, tr);
+    addAsLog(SentryLogLevel.ERROR, msg, tr);
     return Log.e(tag, msg, tr);
   }
 
   public static int wtf(@Nullable String tag, @Nullable String msg) {
     addAsBreadcrumb(tag, SentryLevel.ERROR, msg);
+    addAsLog(SentryLogLevel.FATAL, msg, null);
```

Comment:
i'm not too sure about the log level here.
should we use error instead of fatal?

Reply:
Seems like a good idea to me to map the highest level to FATAL so customers can also query for them separately in Sentry.